packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/property-table/extract-enriched-property-types-from-entity.ts

```diff
@@ -24,22 +30,26 @@ const getDataTypesOfPropertyType = (
 export const extractEnrichedPropertyTypesFromEntity = (
   entity: EntityResponse,
 ): EnrichedPropertyType[] => {
-  return Object.keys(entity.properties).map((propertyTypeId) => {
-    const { propertyType } =
-      entity.entityTypeRootedSubgraph.referencedPropertyTypes.find((val) =>
-        val.propertyTypeId.startsWith(propertyTypeId),
-      ) ?? {};
-
-    if (!propertyType) {
-      throw new Error("propertyType not found");
+  return Object.keys(entity.properties).map((propertyTypeBaseUri) => {
+    const propertyTypeVersions = getPropertyTypesByBaseUri(
```

Comment:
nit: 
we could perhaps make use of destructuring out `const [propertyTypeVersion] = ..` and not have to do the `[0]!` later down. 
The if/throw would still be the same, just on the new destructured value instead.

Reply:
I like it, done.